cmd/ui/main.go

```diff
@@ -2,49 +2,51 @@ package main
 
 import (
 	"embed"
-	"encoding/json"
+	"flag"
 	"io/fs"
 	"net/http"
+	"net/http/httputil"
+	"net/url"
 	"os"
 	"path/filepath"
 
 	"github.com/sirupsen/logrus"
 )
 
+var log = logrus.New()
+
 func main() {
-	log := logrus.New()
+	apiUrlFlag := flag.String("api-url", "http://localhost:8000", "The URL for the applications API server")
+
 	mux := http.NewServeMux()
 
 	mux.Handle("/health/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-	}))
-
-	mux.Handle("/api/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		res := struct {
-			Ok bool `json:"ok"`
-		}{
-			Ok: true,
-		}
-
-		b, err := json.Marshal(res)
+		_, err := w.Write([]byte("ok"))
 
 		if err != nil {
-			log.Errorf("could not marshal: %s", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-
-		if _, err := w.Write(b); err != nil {
-			log.Errorf("error writing bytes: %s", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
+			log.Error(err.Error())
 		}
 	}))
 
 	assetFS := getAssets()
 	assetHandler := http.FileServer(http.FS(assetFS))
 	redirector := createRedirector(assetFS, log)
 
+	apiUrl, err := url.Parse(*apiUrlFlag)
+
+	if err != nil {
+		log.Errorf("could not parse proxy url")
+		os.Exit(1)
+		return
+	}
+
+	log.Infof("api proxy url set to %s", apiUrl.String())
+
+	mux.Handle("/v1/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		proxy(apiUrl, w, r)
+	}))
```

Comment:
Can we call
```golang
if err := pb.RegisterApplicationsHandlerServer(ctx, mux, server.NewApplicationsServer()); err != nil {
  return fmt.Errorf("could not register application: %w", err)
}
```
here to mount all the `/v1` routes here onto `mux` and then completely remove `cmd/wego-server/app/app.go`?

I'm not 100% sure if it would work but would be cool if so.

Reply:
@foot How would we specify that only `/v1` routes go to the `server.NewApplicationServer()`? There are static assets that need to be served by this runtime on `/`, ie `index.html` and `main.js`. 

This was my original complaint around `gRPC`. I can mux the API server into the `ui` runtime easily with `twirp`, but its not obvious how to do it with `gRPC`.

Reply:
Also it should be noted that `wego-server` is the API (standalone) runtime, the file you commented in is the UI server.

Reply:
> @foot How would we specify that only /v1 routes go to the server.NewApplicationServer()?

I think `RegisterApplicationsHandlerServer` only adds the routes in the protobuf file.
https://github.com/weaveworks/weave-gitops/pull/309/files#diff-4f1ecbad04de7fc879a574bf0f86a2d8b2558607c54c9fe2a938c103d020238bR56 on

> Also it should be noted that wego-server is the API (standalone) runtime, the file you commented in is the UI server.

Ah hah! Good point, 2 servers. 👌

Reply:
> I think RegisterApplicationsHandlerServer only adds the routes in the protobuf file.

It only accepts a `runtime.ServeMux`, which I cannot append to. I need it to accept a `http.ServeMux` to add `/metrics` or `/health` (for example).